rollup/sync_service/bridge_client.go

```diff
@@ -36,61 +44,106 @@ func newBridgeClient(ctx context.Context, l1Client EthClient, l1ChainId uint64,
 		return nil, fmt.Errorf("unexpected chain ID, expected = %v, got = %v", l1ChainId, got)
 	}
 
-	filterer, err := NewL1MessageQueueFilterer(l1MessageQueueAddress, l1Client)
+	filtererV1, err := NewL1MessageQueueFilterer(l1MessageQueueV1Address, l1Client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize L1MessageQueueV1Filterer, err = %w", err)
+	}
+	filtererV2, err := NewL1MessageQueueFilterer(l1MessageQueueV2Address, l1Client)
 	if err != nil {
-		return nil, fmt.Errorf("failed to initialize L1MessageQueueFilterer, err = %w", err)
+		return nil, fmt.Errorf("failed to initialize L1MessageQueueV2Filterer, err = %w", err)
 	}
 
 	client := BridgeClient{
-		client:                l1Client,
-		confirmations:         confirmations,
-		l1MessageQueueAddress: l1MessageQueueAddress,
-		filterer:              filterer,
+		client:        l1Client,
+		confirmations: confirmations,
+
+		l1MessageQueueV1Address: l1MessageQueueV1Address,
+		filtererV1:              filtererV1,
+
+		l1MessageQueueV2Address: l1MessageQueueV2Address,
+		filtererV2:              filtererV2,
 	}
 
 	return &client, nil
 }
 
 // fetchMessagesInRange retrieves and parses all L1 messages between the
 // provided from and to L1 block numbers (inclusive).
-func (c *BridgeClient) fetchMessagesInRange(ctx context.Context, from, to uint64) ([]types.L1MessageTx, error) {
+func (c *BridgeClient) fetchMessagesInRange(ctx context.Context, from, to uint64) ([]types.L1MessageTx, []types.L1MessageTx, error) {
 	log.Trace("BridgeClient fetchMessagesInRange", "fromBlock", from, "toBlock", to)
 
+	var msgsV1, msgsV2 []types.L1MessageTx
+
 	opts := bind.FilterOpts{
 		Start:   from,
 		End:     &to,
 		Context: ctx,
 	}
-	it, err := c.filterer.FilterQueueTransaction(&opts, nil, nil)
-	if err != nil {
-		return nil, err
+
+	// Query L1MessageQueueV1 if is enabled. We disable querying of L1MessageQueueV1 once L1MessageQueueV2 is enabled,
+	// and we have received the first QueueTransaction event from L1MessageQueueV2.
+	if !c.skipV1L1Messages {
```

Comment:
So after restarting the node, we'll query V1 once (or a few times) before setting `skipV1L1Messages = true` again, but that's not a problem because we won't find any more messages there?